util/socketset/socketset.go

```diff
@@ -15,87 +15,71 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-// Package socketset is in API for creating a select style netlink
-// socket to receive events from the kernel.
+// Package socketset is in API for creating a select style netlink socket to
+// receive events from the kernel.
 package socketset
 
 import (
+	"bytes"
 	"fmt"
 	"os"
 	"path"
+	"strings"
 	"syscall"
 
 	errwrap "github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
 // SocketSet is used to receive events from a socket and shut it down cleanly
-// when asked. It contains a socket for events and a pipe socket to unblock
-// receive on shutdown.
+// when asked. It contains a socket for events and a pipe socket to unblock receive
+// on shutdown.
 type SocketSet struct {
 	fdEvents int
 	fdPipe   int
 	pipeFile string
 }
 
 // NewSocketSet returns a socketSet, initialized with the given parameters.
-func NewSocketSet(groups uint32, file string) (*SocketSet, error) {
-	// make a netlink socket file descriptor
-	fdEvents, err := unix.Socket(unix.AF_NETLINK, unix.SOCK_RAW, unix.NETLINK_ROUTE)
+func NewSocketSet(groups uint32, name string, proto int) (*SocketSet, error) {
+	fdEvents, err := unix.Socket(unix.AF_NETLINK, unix.SOCK_RAW, proto)
 	if err != nil {
 		return nil, errwrap.Wrapf(err, "error creating netlink socket")
 	}
+
 	// bind to the socket and add add the netlink groups we need to get events
 	if err := unix.Bind(fdEvents, &unix.SockaddrNetlink{
 		Family: unix.AF_NETLINK,
 		Groups: groups,
+		Pid:    uint32(os.Getpid()), // set PID to our process
 	}); err != nil {
 		return nil, errwrap.Wrapf(err, "error binding netlink socket")
 	}
 
-	// create a pipe socket to unblock unix.Select when we close
+	// this pipe unblocks unix.Select upon closing
 	fdPipe, err := unix.Socket(unix.AF_UNIX, unix.SOCK_RAW, unix.PROT_NONE)
 	if err != nil {
 		return nil, errwrap.Wrapf(err, "error creating pipe socket")
 	}
+
 	// bind the pipe to a file
 	if err = unix.Bind(fdPipe, &unix.SockaddrUnix{
-		Name: file,
+		Name: name,
 	}); err != nil {
 		return nil, errwrap.Wrapf(err, "error binding pipe socket")
 	}
+
 	return &SocketSet{
 		fdEvents: fdEvents,
 		fdPipe:   fdPipe,
-		pipeFile: file,
+		pipeFile: name,
 	}, nil
 }
 
-// Shutdown closes the event file descriptor and unblocks receive by sending
-// a message to the pipe file descriptor. It must be called before close, and
```

Comment:
See above please, just curious mostly.

Reply:
Oh, I think this is here because we split the commit into one that just moves + exports the code. Then I split us the methods which pushed this down causing the delta.

Reply:
Okay cool :)